packages/mds-compliance-service/repository/index.ts

```diff
@@ -0,0 +1,215 @@
+import { InsertReturning, RepositoryError, ReadWriteRepository } from '@mds-core/mds-repository'
+import { isDefined, NotFoundError, now } from '@mds-core/mds-utils'
+import { UUID } from '@mds-core/mds-types'
+import {
+  ComplianceSnapshotDomainModel,
+  GetComplianceSnapshotsByTimeIntervalOptions,
+  GetComplianceSnapshotOptions
+} from '../@types'
+import { ComplianceSnapshotEntityToDomain, ComplianceSnapshotDomainToEntityCreate } from './mappers'
+import { ComplianceSnapshotEntity, ComplianceSnapshotEntityModel } from './entities/compliance-snapshot-entity'
+import migrations from './migrations'
+
+class ComplianceSnapshotReadWriteRepository extends ReadWriteRepository {
+  public getComplianceSnapshot = async ({
+    compliance_snapshot_id,
+    provider_id,
+    policy_id,
+    compliance_as_of
+  }: GetComplianceSnapshotOptions): Promise<ComplianceSnapshotDomainModel> => {
+    const { connect } = this
+    try {
+      const connection = await connect('ro')
+      if (isDefined(compliance_snapshot_id)) {
+        const entity = await connection.getRepository(ComplianceSnapshotEntity).findOne({
+          where: {
+            compliance_snapshot_id
+          }
+        })
+        if (!entity) {
+          throw new NotFoundError(`ComplianceSnapshot ${compliance_snapshot_id} not found`)
+        }
+        return ComplianceSnapshotEntityToDomain.map(entity)
+      }
+      if (!isDefined(provider_id) || !isDefined(policy_id)) {
+        throw RepositoryError('provider_id and policy_id must be given if compliance_snapshot_id is not given')
+      }
+
+      const p_compliance_as_of = compliance_as_of || now()
```

Comment:
Can you just do a default in the function args? `getComplianceSnapshot = async ({..., compliance_as_of = now()}: GetComplianceSnapshotOptions)...`

Reply:
Didn't realize you could do that inside of a destructuring.